data-reconciliation-app/src/repositories/kv-repository.ts

```diff
@@ -0,0 +1,69 @@
+import * as ccfapp from "@microsoft/ccf-app";
+
+export interface IKeyValueRepository<T> {
+  set(key: string, value: T): T;
+  get(key: string): T;
+  has(key: string): boolean;
+  keys(): string[];
+  values(): T[];
+  get size(): number;
+}
+
+// generic key-value repository wrapping ccf TypedKvMap interaction
+export class KeyValueRepository<T> implements IKeyValueRepository<T> {
+  private kvStore: ccfapp.TypedKvMap<string, T>;
+
+  public constructor() {
+    this.kvStore = this.getDataMap();
+    console.log(this.kvStore);
+  }
+
+  // update key-value pair or create new record if key not exists
+  public set(key: string, value: T): T {
+    this.kvStore.set(key, value);
+    return value;
+  }
+
+  // retrieve key value from kv-store
+  public get(key: string): T {
+    return this.kvStore.get(key);
+  }
+
+  // check if key exists
+  public has(key: string): boolean {
+    return this.kvStore.has(key);
+  }
+
+  // retrieve all keys of kv-store
+  public keys(): string[] {
+    const keys: string[] = [];
+    this.kvStore.forEach((val, key) => {
+      keys.push(key);
+    });
+    return keys;
+  }
+
+  // retrieve all values of kv-store
+  public values(): T[] {
+    const values: T[] = [];
+    this.kvStore.forEach((val, key) => {
```

Comment:
The repository should handle any exceptions thrown from `kvStore` and return `ServiceResult`